packages/util/src/error.ts

```diff
@@ -6,3 +6,24 @@ export function errorWithCause(message: string, cause: Error): Error {
   }
   return newError;
 }
+
+export class ErrorWithCause extends Error {
+  constructor(message: string, public cause?: Error) {
+    super(message); // 'Error' breaks prototype chain here
+    Object.setPrototypeOf(this, new.target.prototype); // restore prototype chain
+    this.name = 'ErrorWithCause';
```

Comment:
A link to more information on the prototype problems would be helpful here

Reply:
Done.